packages/react-components/react-input/src/components/Input/Input.types.ts

```diff
@@ -9,7 +9,7 @@ export type InputSlots = {
    * The root slot receives the `className` and `style` specified directly on the `<Input>`.
    * All other top-level native props will be applied to the primary slot, `input`.
    */
-  root: NonNullable<Slot<'span'>>;
+  root: NonNullable<Slot<'label'>>;
```

Comment:
I think this is a breaking API change. Putting a block on this PR while we consider the issue and how to handle it.

Reply:
I agree with @spmonahan, I don't think we can merge this change. Doing this would add the text in the contentBefore & contentAfter slots to the input's accessible name, which we almost certainly don't want.

The only way to override it would be via `aria-label` or `aria-labelledby`; using a `<label for="id">` wouldn't be enough. Since in v9 we want to support using built-in HTML accessibility features, I don't think we want to introduce a change that would force people to always use aria to label Inputs.

Reply:
That is the first thing that I thought if if effects others. we can handle when clicking on span also focuses on input. if you have any idea I will follow. thanks

Reply:
@vicheanath A better approach is to adjust the styles in `useInputStyle.styles.ts`. Something along the lines of this:

https://stackblitz.com/edit/stackblitz-starters-umxyha?description=React%20%20%20TypeScript%20starter%20project&file=src%2Findex.tsx,src%2FApp.tsx&title=React%20Starter

Also need to account for before/after slots.

Are you interested in making these changes?

Reply:
i will

Reply:
@spmonahan I have a complete change, if it still has any problem kindly feedback here.